src/plugins/auctionHouseModule/createBid.ts

```diff
@@ -180,19 +183,37 @@ export const createBidBuilder = (
   const bookkeeper: Signer = params.bookkeeper ?? metaplex.identity();
   const receipt = findBidReceiptPda(buyerTradeState);
 
-  return (
-    TransactionBuilder.make<CreateBidBuilderContext>()
-      .setContext({
-        buyerTradeState,
-        tokenAccount,
-        metadata,
-        buyer: toPublicKey(buyer),
-        receipt,
-        bookkeeper: bookkeeper.publicKey,
-        price,
-        tokens,
-      })
+  const builder = TransactionBuilder.make<CreateBidBuilderContext>()
+    .setContext({
+      buyerTradeState,
+      tokenAccount,
+      metadata,
+      buyer: toPublicKey(buyer),
+      receipt,
+      bookkeeper: bookkeeper.publicKey,
+      price,
+      tokens,
+    })
+
+  // Create a TA for public bid if it doesn't exist
+  if (!tokenAccount) {
+    try {
+      await metaplex
+        .tokens()
+        .findTokenByAddress(buyerTokenAccount)
+        .run();
+    } catch {
```

Comment:
Instead of a try/catch, could you use the following?

```ts
const account = await metaplex.rpc().getAccount(buyerTokenAccount)
if (!account.exists) {
  // ...
}
```

I think it reads better than a try/catch in this scenario.